research-software-directory/RSD-as-a-service
# 759 find contributor

# Reduce duplicate entries of contributor and team members in search

Closes #759

Changes proposed in this pull request:
* Search results for contributor or team member are aggregated by ORCID into 1 item. During the process the user can select from available values for avatar, email, role and affiliation (see image 1)
* Additional section, RSD contributors, is added to admin pages. In this section the admin can search for a "person" entries in software and projects. The information can be directly updated. After testing this branch with real RSD data it become clear that we will need to correct significant number of contributor and team member entries. This page can be used to add ORCID id's to all entries of same person. (see image 2)
* Moved created rpc's to separate SQL script "103-person-views.sql" to avoid merge conflicts in the future.

How to test:
* `make start` to build app and create test data
* login as rsd_admin
* edit random software, add same contributor multiple times, for example add Jason Maassen 5 times with different image, email, role and affiliation. 
* edit random project, add Jason Maassen again. In the modal you should have multiple options for image, email, role and affiliation which were previously added to software. As long as all entries you did have same ORCID it  should result into one search result showing all these options.
* Now add new person, again Jason Maassen BUT without ORCID. This should produce two search results. 
* To correct this, [navigate to rsd contributors page of admin section](http://localhost/admin/rsd-contributors)
* Search for `Maassen`. The list of all entries should be shown. Copy ORCID from one of the entries and paste it into the entry which is missing ORCID. You can edit cell by clicking on it. The value is saved after the cell loses the focus.

## Find countributor / team member
![image](https://user-images.githubusercontent.com/9204081/228941552-41443bdb-3ffc-4cf9-b396-84e84c69aa33.png)

## Edit search option modal (after adding Jason Maassen 5 times)
![image](https://user-images.githubusercontent.com/9204081/228942594-e5ecc2fb-6239-44b4-97ba-bb72ef309fdd.png)

## Admin - RSD contributors (edit)
![image](https://user-images.githubusercontent.com/9204081/228944470-c350828a-01ee-4a98-a123-341d17f0e1bb.png)

PR Checklist:
